### DISCUSSION TITLE
httpx ignores no_proxy and https_proxy in docker-compose

### DISCUSSION THREAD

**Discussion by lsch0lz** [encode/httpx#2710] [General]
When I use ``httpx`` inside a Docker container that was built with the ```http_proxy``` and ```http_proxy``` environment variables, I cannot ignore these variables when starting the container via docker-compose.
I set the ```no_proxy``` variable when starting the container so that the container should ignore any proxies it was built with.
```httpx``` still uses those proxies. To solve this I had to set ```trust_env=False``` in my ```AsyncClient```.

Is this expected behaviour? I thought that disabling the proxies in the docker-compose file would automatically tell ```httpx``` to ignore them when making a request.

**Comment by karpetrosyan** [comment]
Hi, can you provide the 'docker-compose.yml' file?
